server/aesdsocket.c

```diff
@@ -563,41 +495,28 @@ int main(int argc, char *argv[])
                 server_socket_state = STATE_EXIT;
                 break;
             }
-            server_socket_state = STATE_OPENING_FILE;
-            break;
-
-        case STATE_OPENING_FILE:
-
-            LOG_DBG("STATE_OPENING_FILE\n");
-            file_descriptor = open("/var/tmp/aesdsocketdata", O_CREAT | O_RDWR | O_TRUNC, 0644);
+            LOG_DBG("Creating a head for socket threads linked list\n");
+            SLIST_HEAD(slisthead, slist_data_s)
+            socket_head;
+            SLIST_INIT(&socket_head);
+            server_socket_state = STATE_ACCEPTING;
+#ifndef USE_AESD_CHAR_DRIVER
```

Comment:
@rajatchaple as noted above, I'm not sure why the logic needs to be different here.  A comment about this would be good.